docs/create-a-business-component.md

```diff
@@ -0,0 +1,345 @@
+> Just know that you could use
+> [Error Boundaries](https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html)
+> for that.
+
+```js
+// src/theme/modules/StoreLocator/StoreLocator.js;
+
+import React from "react";
+import PropTypes from "prop-types";
+import { Map, TileLayer, Marker, Popup } from "react-leaflet";
+import "leaflet/dist/leaflet.css";
+import "leaflet/dist/images/marker-icon.png";
+import "leaflet/dist/images/marker-shadow.png";
+import StoreLocatoreQuery from "./StoreLocatorQuery.gql";
+import EnhanceStoreLocator from "./EnhanceStoreLocator";
+
+const StoreLocator = props => {
+  if (props.loading) {
+    return <div>Loading...</div>;
+  }
+
+  const coordinates = [
+    props.store.coordinates.longitude,
+    props.store.coordinates.latitude
+  ];
+  const defaultZoom = 14;
+
+  return (
+    <div>
+      <Map
+        center={coordinates}
+        zoom={defaultZoom}
+        style={{ height: "600px", width: "800px" }}
+      >
+        <TileLayer
+          attribution="&amp;copy <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        />
+        <Marker position={coordinates}>
+          <Popup>
+            <div>
+              My awesome store ─ {props.store.name}
+              Email: {props.store.owner.email}
+              Phone: {props.store.phone}
+            </div>
+          </Popup>
+        </Marker>
+      </Map>
+    </div>
+  );
+};
+
+StoreLocator.propTypes = {
+  loading: PropTypes.bool.isRequired,
+  store: PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    phone: PropTypes.string.isRequired,
+    owner: PropTypes.shape({
+      email: PropTypes.string.isRequired
+    }).isRequired,
+    coordinates: PropTypes.shape({
+      longitude: PropTypes.number.isRequired,
+      latitude: PropTypes.number.isRequired
+    }).isRequired
+  })
+};
+
+export default EnhanceStoreLocator(StoreLocatoreQuery)(StoreLocator);
+```
+
+> A side note on **Protypes**. The Proptypes allow us to validate the props
+> passed to a child component, this will avoid breaking your app passing invalid
+> props.  
+This will also serve as a props documentation for the other developers
+> in your team. They may be a little verbose, but it is important to maintain
+> them updated.
+
+## Using it in our App
+
+We now have all the required pieces, to finish, we just have to import our brand
+new shiny module and use it in the home page.
+
+```js
+// src/theme/pages/Home/Home.js;
+//...
+import StoreLocator from "../../modules/StoreLocator";
+
+const Home = ({ store }) => (
+  <div className="page page--home">
+    //...
+    <StoreLocator />
+  </div>
+);
+```
+
+> The store locator we just created is very simple and it has a very limited
+> Business impact. It was for the sake of this guide. A component such as this
+> one might have its place in the ui directory in your application.
```

Comment:
I think the last sentence may be misleading, since in the beginning we explained why it should live in the `modules` directory.

We could remove it, and only emphasize on the fact that it is a simple and naive implementation to illustrate how to create such components. Splitting code accordingly needs practice and refinement, thus one team could split code differently.

If we want to explain what should go to UI, then there needs to be a clear note about not moving GraphQL code there or things related to the store. But I feel like it would be too much information here.